bpf/lbmap/lbmap.go

```diff
@@ -173,13 +225,95 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+// gcd computes the gcd of two numbers.
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+// Generates a wrr sequence based on provided weights.
+func GenerateWrrSeq(weights []uint16) *RRSeqValue {
+	svcRRSeq := RRSeqValue{}
+
+	n := len(weights)
+	if n < 2 {
+		return nil
+	}
+
+	g := uint16(0)
+	for i := 0; i < n; i++ {
+		if weights[i] != 0 {
+			g = gcd(g, weights[i])
+		}
+	}
+
+	// This means all the weights are 0.
+	if g == 0 {
+		return nil
```

Comment:
.... which allows you to return meaningful errors here to indicate to the user why generation has failed:
`return nil, fmt.Errorf("all specified weights are 0")`